SimplenoteTests/NoticeControllerTests.swift

```diff
@@ -0,0 +1,238 @@
+import XCTest
+@testable import Simplenote
+
+class NoticeControllerTests: XCTestCase {
+    private lazy var presenter = MockNoticePresenter()
+    private lazy var timerFactory = MockTimerFactory()
+    private lazy var controller = NoticeController(presenter: presenter, timerFactory: timerFactory)
+
+    func testSetupNoticeContoller() {
+        controller.setupNoticeController()
+
+        XCTAssertTrue(presenter.listeningToKeyboard == true)
+    }
+
+    func testPresentNotice() {
+        let notice = Notice(message: "Message", action: nil)
+        let expectedActions: [MockNoticePresenter.Action] = [
+            .present(notice.message)
+        ]
+        controller.present(notice)
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+        XCTAssertEqual(presenter.lastNoticeView?.message, notice.message)
+    }
+
+    func testControlDoesNotPresentNewNoticeIfPresenting() {
+        let noticeA = Notice(message: "Message A", action: nil)
+        let noticeB = Notice(message: "Message B", action: nil)
+        let expectedActions: [MockNoticePresenter.Action] = [
+            .present(noticeA.message)
+        ]
+
+        controller.present(noticeA)
+        controller.present(noticeB)
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+        XCTAssertEqual(presenter.lastNoticeView?.message, noticeA.message)
+    }
+
+    func testAppendToQueueIfNew() {
+        let noticeA = Notice(message: "Message A", action: nil)
+        let noticeB = Notice(message: "Message B", action: nil)
+
+        controller.present(noticeA)
+        XCTAssertEqual(controller.pendingNotices, 0)
+
+        controller.present(noticeA)
+        XCTAssertEqual(controller.pendingNotices, 0)
+
+        controller.present(noticeB)
+        XCTAssertEqual(controller.pendingNotices, 1)
+
+        controller.present(noticeB)
+        XCTAssertEqual(controller.pendingNotices, 1)
+    }
```

Comment:
Do you think we can also use actionLog for this and get rid of `pendingNotices`?

Reply:
Had to think about that one for a little bit, but yes I think we can!